bin/generate.dart

```diff
@@ -130,17 +127,52 @@ class DiagramGenerator {
       filters.add('--name');
       filters.add(path.basenameWithoutExtension(name));
     }
-    final List<String> filterArgs = filters.isNotEmpty
-        ? <String>['--route', 'args:${Uri.encodeComponent(filters.join(' '))}']
-        : <String>[];
+    if (deviceTargetPlatform.startsWith('android')) {
+      filters.add('--platform');
+      filters.add('android');
+    } else if (deviceTargetPlatform.startsWith('ios')) {
+      filters.add('--platform');
+      filters.add('ios');
+    } else if (deviceTargetPlatform.startsWith('darwin')) {
+      filters.add('--platform');
+      filters.add('macos');
+    } else if (deviceTargetPlatform.startsWith('linux')) {
+      filters.add('--platform');
+      filters.add('linux');
+    } else if (deviceTargetPlatform.startsWith('windows')) {
+      filters.add('--platform');
+      filters.add('windows');
+    } else if (deviceTargetPlatform.startsWith('fuchsia')) {
+      filters.add('--platform');
+      filters.add('fuchsia');
+    } else {
+      throw GeneratorException('Unsupported target platform $deviceTargetPlatform for device $deviceId');
+    }
+    filters.add('--output-dir');
+    filters.add(temporaryDirectory.absolute.path);
+    late final List<String> filterArgs;
+    if (deviceTargetPlatform.startsWith('android')) {
+      filterArgs = filters.isNotEmpty
+          ? <String>[
+              '--route',
+              'args:${Uri.encodeComponent(filters.join(' '))}'
+            ]
+          : <String>[];
+    } else {
+      filterArgs = <String>[];
+      for (final String arg in filters) {
+        filterArgs.add('--dart-entrypoint-args');
+        filterArgs.add(arg);
```

Comment:
we really should just support --route everywhere... (but that's an issue for another PR)